argo/providers/vai/provider.go

```diff
@@ -193,8 +202,38 @@ func (vaip VAIProvider) DeletePipeline(ctx context.Context, providerConfig VAIPr
 	return nil
 }
 
+func ExtractFromMap[T any](targetMap map[string]any, fieldName string) (result T, err error) {
+	result, ok := targetMap[fieldName].(T)
+	if !ok {
+		err = fmt.Errorf("failed extracting field %s of type %T from the given map", fieldName, result)
+	}
+	return result, err
+}
+
+func extractPipelineNameFromSpec(pipelineSpec map[string]any) (string, error) {
+
+	pipelineInfo, err := ExtractFromMap[map[string]any](pipelineSpec, "pipelineInfo")
+	if err != nil {
+		return "", err
+	}
+	runId, err := ExtractFromMap[string](pipelineInfo, "name")
+	if err != nil {
+		return "", err
+	}
+
+	return runId, nil
+}
+
+func extractRunNameSuffix(pipelineName string) (string, error) {
```

Comment:
I can't see where this is used apart from the tests 🙈

Reply:
At least it was tested 🐵